packages/terra-dropdown-button/src/DropdownButton.jsx

```diff
@@ -0,0 +1,133 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import KeyCode from 'keycode-js';
+import DropdownButtonBase from './_DropdownButtonBase';
+import styles from './DropdownButton.module.scss';
+import Button from './_Button';
+import SplitButton, { Variants as SplitButtonVariants } from './SplitButton';
+
+const cx = classNames.bind(styles);
+
+const Variants = {
+  NEUTRAL: 'neutral',
+  EMPHASIS: 'emphasis',
+  GHOST: 'ghost',
+};
+
+const propTypes = {
+  /**
+   * The options to display in the dropdown. Should be of type `Button`.
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Determines whether the dropdown should be disabled.
+   */
+  disabled: PropTypes.bool,
+  /**
+   * Determines whether the component should have block styles applied. The dropdown will match the component's width.
+   */
+  isBlock: PropTypes.bool,
+  /**
+   * Sets the text that will be shown on the dropdown button.
+   */
+  label: PropTypes.string.isRequired,
+  /**
+   * Sets the styles of the component.
+   * Must be one of 'neutral', 'emphasis', or 'ghost'.
+   */
+  variant: PropTypes.oneOf(Object.values(Variants)),
+};
+
+const defaultProps = {
+  disabled: false,
+  isBlock: false,
+  variant: 'neutral',
+};
+
+class DropdownButton extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.handleDropdownButtonClick = this.handleDropdownButtonClick.bind(this);
+    this.handleDropdownRequestClose = this.handleDropdownRequestClose.bind(this);
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+    this.handleKeyUp = this.handleKeyUp.bind(this);
+
+    this.state = { isOpen: false, active: false };
+  }
+
+  handleDropdownButtonClick() {
+    this.setState({ isOpen: true });
+  }
+
+  handleDropdownRequestClose() {
+    this.setState({ isOpen: false });
+  }
+
+  /*
+    In FireFox active styles don't get applied on space
+   */
+  handleKeyDown(event) {
+    if (event.keyCode === KeyCode.KEY_SPACE) {
```

Comment:
Do we need to toggle focus state for tabbing to and from this component?

Reply:
Also, do we need to handle `enter` for opening the dropdown as well?

Reply:
Focus styles seem to appear when tabbed to in Firefox and Chrome.

When pressing `enter` on the buttons containing chevrons `onClick` gets called immediately which opens the dropdown. Active styles are applied to the chevron button while the dropdown is open.